### PR TITLE
For #20919 add confirm dialog when turning on/off experiments

### PR DESCRIPTION
**Pull request by Amejia481**
As requested by QA we want to should confirmation dialog instead of a Toast, for more context see https://github.com/mozilla-mobile/fenix/issues/20919#issuecomment-904613713

https://user-images.githubusercontent.com/773158/130671513-78022246-cac6-41d1-9046-72439cb075f9.mp4

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/settings/studies/StudiesView.kt [hunk 1]
```diff
@@ -138,20 +173,6 @@ class StudiesView(
...
-            {
-                exitProcess(0)
-            },
-            OVERRIDE_EXIT_DELAY
-        )
-    }
-
-    companion object {
-        private const val OVERRIDE_EXIT_DELAY = 3000L
+        exitProcess(0)
```
Comment (eliserichards) [inline app/src/main/java/org/mozilla/fenix/settings/studies/StudiesView.kt:176]: We no longer need the `post` for this?

**Review by eliserichards — APPROVED** [review]
Looks good!

File: app/src/main/java/org/mozilla/fenix/settings/studies/StudiesView.kt (hunk 1 above)
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/settings/studies/StudiesView.kt:176]: We used the `postDelayed` function to delay  killing the app `exitProcess(0)` and giving some time to the user to read the toast message :)

File: app/src/main/java/org/mozilla/fenix/settings/studies/StudiesView.kt [hunk 2]
```diff
@@ -47,15 +47,40 @@ class StudiesView(
...
         provideStudiesSwitch().isChecked = settings.isExperimentationEnabled
-        provideStudiesSwitch().setOnCheckedChangeListener { _, isChecked ->
-            settings.isExperimentationEnabled = isChecked
-            experiments.globalUserParticipation = isChecked
-            provideStudiesTitle().text = getSwitchTitle()
-            quitTheApp()
+        provideStudiesSwitch().setOnClickListener {
+            val isChecked = provideStudiesSwitch().isChecked
+            provideStudiesTitle().text = getSwitchCheckedTitle()
+            val builder = AlertDialog.Builder(context)
```
Comment (MozillaNoah) [inline app/src/main/java/org/mozilla/fenix/settings/studies/StudiesView.kt:57]: I'm kinda surprised we don't have Dialogs in AC

File: app/src/main/java/org/mozilla/fenix/settings/studies/StudiesView.kt [hunk 3]
```diff
@@ -127,6 +152,16 @@ class StudiesView(
...
     }
 
+    @VisibleForTesting
+    internal fun getSwitchCheckedTitle(): String {
+        val stringId = if (provideStudiesSwitch().isChecked) {
+            R.string.studies_on
+        } else {
+            R.string.studies_off
+        }
+        return context.getString(stringId)
```
Comment (MozillaNoah) [inline app/src/main/java/org/mozilla/fenix/settings/studies/StudiesView.kt:162]: This IF could alternatively be inside the getString

File: app/src/main/java/org/mozilla/fenix/settings/studies/StudiesView.kt (hunk 2 above)
Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/settings/studies/StudiesView.kt:57]: There isn't any useful functionality that we could add in AC which the Android system APIs already provide. We could have _Fenix_ dialog wrappers to ensure consistent styling or UX requirements; that could work.

**Pull request merged**